packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -18,8 +21,14 @@ class SharedPreferences {
   static const String _prefix = 'flutter.';
   static Completer<SharedPreferences> _completer;
 
-  static SharedPreferencesStorePlatform get _store =>
-      SharedPreferencesStorePlatform.instance;
+  static SharedPreferencesStorePlatform get _store {
+    // This is a workaround until https://github.com/flutter/flutter/issues/52267
+    // is fixed. There is no way currently to register a Dart-only plugin.
+    if (!kIsWeb && LocalPlatform().isLinux) {
+      return SharedPreferencesLinux();
```

Comment:
Should this be a singleton like `SharedPreferencesStorePlatform.instance`? I copied this from https://github.com/flutter/plugins/pull/2631.

Reply:
Since it's stateful, yes, it'll need to be a singleton.